lib/miq_expression/field.rb

```diff
@@ -17,6 +17,10 @@ def self.parse!(field)
     parse(field) or raise ParseError, field
   end
 
+  def self.filter_has_many(cols)
+    cols ? cols.select { |_, column| parse(column).try(:plural?) } : []
```

Comment:
yes that's right  format of `cols` is very ui/reports specific,so I'll move to the controller repo.